lib/vision/index.js

```diff
@@ -0,0 +1,1376 @@
+ * converted to `false`.
+ *
+ *   - **False**
+ *   - `VERY_UNLIKELY`
+ *   - `UNLIKELY`
+ *   - `POSSIBLE`
+ *   - **True**
+ *   - `LIKELY`
+ *   - `VERY_LIKELY`
+ *
+ * The API will also return many values represented in a `[0,1]` range. We
+ * convert these to a `[0,100]` value. E.g, `0.4` is represented as `40`.
+ *
+ * For the response in the original format, review the `apiResponse` argument
+ * your callback receives.
+ *
+ * @param {string|string[]|module:storage/file|module:storage/file[]} images -
+ *     The source image(s) to run the detection on. It can be either a local
+ *     image path or a gcloud File object.
+ * @param {string[]|object} config - An array of types or a configuration
+ *     object.
+ * @param {string[]} config.types - An array of feature types to detect from the
+ *     provided images. Acceptable values: `face`, `landmark`, `label`, `logo`,
+ *     `properties`, `safeSearch`, `text`.
+ * @param {boolean=} config.verbose - Use verbose mode, which returns a less-
+ *     simplistic representation of the annotation (default: `false`).
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {object|object[]} callback.detections - If a single detection type was
+ *     asked for, it will be returned in its raw form; either an object or array
+ *     of objects. If multiple detection types were requested, you will receive
+ *     an object with keys for each detection type (listed above in
+ *     `config.types`). Additionally, if multiple images were provided, you will
+ *     receive an array of detection objects, each representing an image. See
+ *     the examples below for more information.
+ * @param {object} callback.apiResponse - Raw API response.
+ *
+ * @example
+ * var types = [
+ *   'face',
+ *   'label'
+ * ];
+ *
+ * vision.detect('image.jpg', types, function(err, detections, apiResponse) {
+ *   // detections = {
+ *   //   face: [...],
+ *   //   label: [...]
+ *   // }
+ * });
+ *
+ * //-
+ * // Supply multiple images for feature detection.
+ * //-
+ * var images = [
+ *   'image.jpg',
+ *   'image-two.jpg'
+ * ];
+ *
+ * var types = [
+ *   'face',
+ *   'label'
+ * ];
+ *
+ * vision.detect(images, types, function(err, detections, apiResponse) {
+ *   // detections = [
+ *   //   // Detections for image.jpg:
+ *   //   {
+ *   //     face: [...],
+ *   //     label: [...]
+ *   //   },
+ *   //
+ *   //   // Detections for image-two.jpg:
+ *   //   {
+ *   //     face: [...],
+ *   //     label: [...]
+ *   //   }
+ *   // ]
+ * });
+ */
+Vision.prototype.detect = function(images, options, callback) {
+  var self = this;
+  var isSingleImage = !is.array(images) || images.length === 1;
+
+  if (!is.object(options)) {
+    options = {
+      types: options
+    };
+  }
+
+  var types = arrify(options.types);
+
+  var typeShortNameToFullName = {
+    face: 'FACE_DETECTION',
```

Comment:
Since we're always returning arrays, what are your feelings on dropping the singular nouns? I saw an assertion that checked for `detections.face` as an Array, which threw me off a little.